onecodex/models/collection.py

```diff
@@ -630,15 +654,21 @@ def to_otu(self, biom_id=None):
         otu["shape"] = [num_rows, num_cols]
         otu["data"] = []
 
-        for present_taxa in sorted(rows):
-            # add the row entry
+        for tax_id in sorted(rows):
             row_id = len(otu["rows"])
-            otu["rows"].append(
-                {"id": present_taxa, "metadata": {"taxonomy": tax_ids_to_names[present_taxa]}}
-            )
+            tax_node = root.find(tax_id)
```

Comment:
it'd be nice to not call `root.find()` twice for each `tax_id`

Reply:
skbio is caching subsequent calls to `find` so this should be just as fast as keeping a separate map of `tax_id -> node`